### PR TITLE
Discard disconnected compiler workers from the pool queue

### PR DESCRIPTION
**Pull request by fantix**
This was a bug in the compiler pool that disconnected workers were never removed from the queue, that concurrent compilation may run into the disconnected worker and trigger an error.

### CODE REVIEW

File: edb/server/compiler_pool/pool.py [hunk 1]
```diff
@@ -452,6 +452,19 @@ def sync_worker_state_cb(
 
         return preargs, callback
 
+    async def _acquire_worker(self):
+        while (
+            worker := await self._workers_queue.acquire()
+        ).get_pid() not in self._workers:
+            # The worker was disconnected; skip to the next one.
+            pass
+        return worker
```
Comment (elprans) [inline edb/server/compiler_pool/pool.py:461]: Can all workers in the queue be disconnected?  If so, then this potentially still leaks a disconnected worker (the last one).

File: edb/server/compiler_pool/pool.py [hunk 2]
```diff
@@ -452,6 +452,19 @@ def sync_worker_state_cb(
 
         return preargs, callback
 
+    async def _acquire_worker(self):
+        while (
+            worker := await self._workers_queue.acquire()
```
Comment (1st1) [inline edb/server/compiler_pool/pool.py:457]: Can this loop complete with no connected worker found?

File: edb/server/compiler_pool/pool.py (hunk 1 above)
Comment (fantix) [reply edb/server/compiler_pool/pool.py:461]: > Can all workers in the queue be disconnected?

Yes!

> this potentially still leaks a disconnected worker (the last one).

But the last one in the queue won't escape the `.get_pid() not in self._workers` check? `_workers_queue.acquire()` will block after that, until a worker whose pid is in `self._workers` is found 🤔

File: edb/server/compiler_pool/pool.py (hunk 2 above)
Comment (fantix) [reply edb/server/compiler_pool/pool.py:457]: I don't think so, it'll block until a connected one is found I think.

File: edb/server/compiler_pool/pool.py (hunk 1 above)
Comment (elprans) [reply edb/server/compiler_pool/pool.py:461]: Oh, right. `acquire()` blocks.

**Review by 1st1 — APPROVED** [review]

**Pull request merged**